dialogue-core/src/main/java/com/palantir/dialogue/core/Channels.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.dialogue.Channel;
+
+final class Channels {
```

Comment:
This is a big part of what I was trying to get at here - I wanted there to be a single place where we compose all the channels.

Reply:
#688 pulls out the channel updating logic from DialogueChannel which I think results in DialogueChannel being the sole place that does the composition